client/network/src/protocol/notifications/handler.rs

```diff
@@ -1011,596 +1012,598 @@ pub mod tests {
-		})
-		.await;
-	}
-
-	#[tokio::test]
-	async fn close_desired_by_remote() {
-		let mut handler = notifs_handler();
-		let (io, io2) = MockSubstream::negotiated().await;
-		let mut codec = UviBytes::default();
-		codec.set_max_len(usize::MAX);
-
-		let notif_in = NotificationsInOpen {
-			handshake: b"hello, world".to_vec(),
-			negotiated_fallback: None,
-			substream: NotificationsInSubstream::new(
-				Framed::new(io, codec),
-				NotificationsInSubstreamHandshake::PendingSend(vec![1, 2, 3, 4]),
-			),
-		};
-
-		// add new inbound substream but close it immediately and verify that correct events are
-		// emitted
-		handler.on_connection_event(handler::ConnectionEvent::FullyNegotiatedInbound(
-			handler::FullyNegotiatedInbound { protocol: (notif_in, 0), info: () },
-		));
-		drop(io2);
-
-		futures::future::poll_fn(|cx| {
-			assert!(std::matches!(
-				handler.poll(cx),
-				Poll::Ready(ConnectionHandlerEvent::Custom(
-					NotifsHandlerOut::OpenDesiredByRemote { protocol_index: 0 },
-				))
-			));
-			assert!(std::matches!(
-				handler.poll(cx),
-				Poll::Ready(ConnectionHandlerEvent::Custom(NotifsHandlerOut::CloseDesired {
-					protocol_index: 0
-				},))
-			));
-			Poll::Ready(())
-		})
-		.await;
-	}
+	// FIXME: find a way to rewrite these tests.
+
+	// /// Create new [`NotifsHandler`].
+	// fn notifs_handler() -> NotifsHandler {
+	// 	let proto = Protocol {
+	// 		config: ProtocolConfig {
+	// 			name: "/foo".into(),
+	// 			fallback_names: vec![],
+	// 			handshake: Arc::new(RwLock::new(b"hello, world".to_vec())),
+	// 			max_notification_size: u64::MAX,
+	// 		},
+	// 		in_upgrade: NotificationsIn::new("/foo", Vec::new(), u64::MAX),
+	// 		state: State::Closed { pending_opening: false },
+	// 	};
+
+	// 	NotifsHandler {
+	// 		protocols: vec![proto],
+	// 		when_connection_open: Instant::now(),
+	// 		endpoint: ConnectedPoint::Listener {
+	// 			local_addr: Multiaddr::empty(),
+	// 			send_back_addr: Multiaddr::empty(),
+	// 		},
+	// 		peer_id: PeerId::random(),
+	// 		events_queue: VecDeque::new(),
+	// 	}
+	// }
+
+	// 	// verify that if another substream is attempted to be opened by remote while an inbound
+	// 	// substream already exists, the new inbound stream is rejected and closed by the local node.
+	// 	#[tokio::test]
+	// 	async fn second_open_desired_by_remote_rejected() {
+	// 		let mut handler = notifs_handler();
+	// 		let (io, mut io2) = MockSubstream::negotiated().await;
+	// 		let mut codec = UviBytes::default();
+	// 		codec.set_max_len(usize::MAX);
+
+	// 		let notif_in = NotificationsInOpen {
+	// 			handshake: b"hello, world".to_vec(),
+	// 			negotiated_fallback: None,
+	// 			substream: NotificationsInSubstream::new(
+	// 				Framed::new(io, codec),
+	// 				NotificationsInSubstreamHandshake::NotSent,
+	// 			),
+	// 		};
+
+	// 		handler.on_connection_event(handler::ConnectionEvent::FullyNegotiatedInbound(
+	// 			handler::FullyNegotiatedInbound { protocol: (notif_in, 0), info: () },
```

Comment:
haven't found a way around this yet...

Reply:
The test can commented out if it's a blocker and we can look into enabling it in the future

Reply:
Could you create an issue so that we remember to look into this issue?

Reply:
I haven't looked into the details of this test but we extensively use https://docs.rs/futures_ringbuf to simulate a network connection in-memory.

Reply:
@thomaseizinger 

IIUC the reason these were commented out was because something in libp2p was made private. What was the exact reason @melekes?